app/models/instance_note.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: instance_notes
+#
+#  id         :bigint(8)        not null, primary key
+#  domain     :string
+#  account_id :bigint(8)
+#  content    :text
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+class InstanceNote < ApplicationRecord
+  include DomainNormalizable
+  include DomainMaterializable
+
+  CONTENT_SIZE_LIMIT = 2_000
+
+  belongs_to :account
+  # belongs_to :instance, inverse_of: :notes
```

Comment:
For some reason this relationship didn't work, and I was getting a validation error. I think it's because `Instance` isn't a real table, but a view or something? Or something to do with there not being an `id` column, and instances are identifier by domain.

Reply:
There are `foreign_key` and `primary_key` options to the association methods for these scenarios, which may be relevant (some usage already on `Instance` for example ... although they are going from `Instance` to other models, rather than from other places to `Instance` like here ... but should be straightforward).